src/main/java/org/jvnet/hudson/plugins/port_allocator/PortAllocator.java

```diff
@@ -49,14 +62,22 @@ public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener l
                 prefPortMap = prevAlloc.getPreviousAllocatedPorts();
             }
         }
+        final PortAllocationManager mpam = PortAllocationManager.getManager(master);
         final PortAllocationManager pam = PortAllocationManager.getManager(cur);
         Map<String,Integer> portMap = new HashMap<String,Integer>();
         final List<Port> allocated = new ArrayList<Port>();
 
         for (PortType pt : ports) {
-            logger.println("Allocating TCP port "+pt.name);
+        	boolean global = false;
+        	try {
+				Pool pool = PortAllocator.DESCRIPTOR.getPoolByName(pt.name);
+				global = pool.global;
+			} catch (PoolNotDefinedException e) {
+				// ignore, global = false
+			}
+            logger.println("Allocating TCP port "+pt.name + (global ? " (global)" : ""));
             int prefPort = prefPortMap.get(pt.name)== null?0:prefPortMap.get(pt.name);
-            Port p = pt.allocate(build, pam, prefPort, launcher, listener);
+            Port p = pt.allocate(build, global ? mpam : pam, prefPort, launcher, listener);
```

Comment:
Hello,

My main concern with this PR is that we are "allocating" ports on slaves only checking its availability in the master. I may have misunderstood the use case, or maybe it not in the scope of this plugin.

Please open a JIRA issue and/or a mailing list thread to elaborate.

Thanks.